langserver/references.go

```diff
@@ -142,7 +157,10 @@ func (h *LangHandler) handleTextDocumentReferences(ctx context.Context, conn JSO
 		log.Printf("info: timeout during references for %s, found %d refs", defpkg, len(locs))
 	}
 
-	if findRefErr != nil {
+	// If we find references then we can ignore findRefErr. It should only
+	// be non-nil due to timeouts or our last findReferences doesn't find
+	// the def.
+	if len(locs) == 0 && findRefErr != nil {
```

Comment:
`findRefErr` can be non-nil (ie it could be `ctx.Err()` or an error typechecking `defpkg`). However, if we found references and it is non-nil that means:
* context timeout, we want partial results.
* we could do findReferences more than once. The last check could fail, but an earlier one succeed. We should still return results in such a case.